tests/test_0692_fsspec.py

```diff
@@ -65,16 +66,36 @@ def test_open_fsspec_s3(handler):
         assert len(data) == 8004
 
 
-@pytest.mark.parametrize("handler", [uproot.source.fsspec.FSSpecSource, None])
-@pytest.mark.skip("you must provide an ssh server to test this")
-def test_open_fsspec_ssh(handler):
+def test_open_fsspec_ssh():
     pytest.importorskip("sshfs")
 
-    # change this to a server you have access to
-    uri = "ssh://user@host:22/tmp/file.root"
-    with uproot.open(uri, handler=handler) as f:
-        data = f["Events/MET_pt"].array(library="np")
-        assert len(data) == 40
+    # check localhost has ssh access to itself
+    try:
+        user = subprocess.check_output(["whoami"]).strip().decode("ascii")
+        host = "localhost"
+        ssh_command = f"ssh {user}@{host} 'echo hello'"
+        result = subprocess.run(
+            ssh_command,
+            shell=True,
+            text=True,
+            capture_output=True,
+        )
```

Comment:
Generally, the list-form of subprocess commands are preferred because they're less susceptible to malicious code injections. I think if someone somehow commandeered the `whoami` command to return "`; rm -rf /*`; ", then the list-form would not allow it to run because individual items in the list can't contain unquoted spaces.

```suggestion
        user = subprocess.check_output(["whoami"]).strip().decode("ascii")
        host = "localhost"
        ssh_command = ["ssh", f"{user}@{host}", "'echo hello'"]
        result = subprocess.run(
            ssh_command,
            shell=True,
            text=True,
            capture_output=True,
        )
```

Reply:
Wow I didn't know that, I always wondered why the recommended way was using a list... now I have an answer (and unlocked a new fear!)